cmd/init.go

```diff
@@ -83,12 +85,29 @@ func init() {
 	rootCmd.AddCommand(initCmd)
 }
 
-func validate(projName string) {
+func getProjectName(pkgName string) (string, error) {
 	validate := validator.New()
-	errs := validate.Var(projName, "required,alphanum")
+
+	errs := validate.Var(pkgName, "max=100,printascii,excludesall=!\"#$%&'()*+0x2C:;<=>?@[\\]^`{0x7C~},startsnotwith=/,startsnotwith=-,startsnotwith=_,startsnotwith=.endsnotwith=/,endsnotwith=-,endsnotwith=_,endsnotwith=.")
```

Comment:
@yu-yk I found this https://go.dev/ref/mod#go-mod-file-ident

Please check the Module paths requirements

Reply:
I think now it already covered most of the case, but I will check their detailed specs.